File: mobile_apps/IOS/user/Edelivery/Models/DataModel/Specifications.swift


import Foundation


public class Specifications {
	public var list : Array<SpecificationListItem>?
	public var type : Int?
	public var is_required : Bool?
	public var name : String?
    public var unique_id:Int?
    public var price:Double?
    var user_can_add_specification_quantity: Bool = false
    
    var range : Int!
    var rangeMax : Int!
    var selectionMessage:String = ""
    var selectedCount = 0
    var nameArray = [String]()
    var isAssociated: Bool = false
    var isParentAssociate: Bool = false
    var modifierName: String = ""
    var modifierGroupName: String = ""
    var modifierId: String = ""
    var modifierGroupId: String = ""
    var _id: String = ""
    
    public class func modelsFromDictionaryArray(array:NSArray) -> [Specifications] {
        var models:[Specifications] = []
        for item in array {
            models.append(Specifications(dictionary: item as! NSDictionary)!)
        }
        return models
    }
	required public init?(dictionary: NSDictionary) {

		if (dictionary["list"] != nil) { list = SpecificationListItem.modelsFromDictionaryArray(array: dictionary["list"] as! NSArray)
        }
        range = (dictionary["range"] as? Int) ?? 0
        _id = (dictionary["_id"] as? String) ?? ""
        rangeMax = (dictionary["max_range"] as? Int) ?? 0
		type = (dictionary["type"] as? Int) ?? 0
		is_required = dictionary["is_required"] as? Bool
        user_can_add_specification_quantity = dictionary["user_can_add_specification_quantity"] as? Bool ?? false
        
        //Userapp //8-7
         if isNotNSNull(object: dictionary["name"] as AnyObject){
                 if dictionary["name"] as? String == nil{
                     if let name = dictionary["name"] as? [Any] {
                         nameArray.removeAll()
                         
                         for obj in name {
                             if let strName = obj as? String {
                                 nameArray.append(strName)
                             } else {
                                 nameArray.append("")
                             }
                         }
                         
                         if nameArray.count > 0{
                             self.name = nameArray[0]
                         }else{
                             self.name = ""
                         }
                     }
                 }else{
                     name = dictionary["name"] as? String ?? ""
                 }
             }else{
                 name = ""
             }
        
//		//name = dictionary["name"] as? String
        unique_id = (dictionary["unique_id"] as? Int) ?? 0
        price = dictionary["price"] as? Double ?? 0.0
        
        modifierId = dictionary["modifierId"] as? String ?? ""
        modifierGroupId = dictionary["modifierGroupId"] as? String ?? ""
        modifierName = dictionary["modifierName"] as? String ?? ""
        modifierGroupName = dictionary["modifierGroupName"] as? String ?? ""
        isAssociated = dictionary["isAssociated"] as? Bool ?? false
        isParentAssociate = dictionary["isParentAssociate"] as? Bool ?? false
	}
    
    public init () {
    }
    
	public func dictionaryRepresentation() -> NSDictionary {
        let dictionary = NSMutableDictionary()
        dictionary.setValue(self.range, forKey: "range")
        dictionary.setValue(self.rangeMax, forKey: "max_range")
        dictionary.setValue(self.type, forKey: "type")
		dictionary.setValue(self.is_required, forKey: "is_required")
        //Userapp
//		dictionary.setValue(self.name, forKey: "name")
        if nameArray.count > 0
        {
            dictionary.setValue(self.nameArray, forKey: "name")
        }else{
            dictionary.setValue(self.name, forKey: "name")
        }
        dictionary.setValue(self.unique_id, forKey: "unique_id")
        dictionary.setValue(self.price, forKey: "price")
        dictionary.setValue(self.user_can_add_specification_quantity, forKey: "user_can_add_specification_quantity")
        
        var myArray:[Any] = []
        for item in self.list! {
            myArray.append(item.dictionaryRepresentation())
        }
        
        dictionary.setValue(myArray, forKey: "list")
        dictionary.setValue(self._id, forKey: "_id")
        
		return dictionary
	}
    
    public func getProductJson() -> NSDictionary {
        let dictionary = NSMutableDictionary()
        dictionary.setValue(self.unique_id, forKey: "unique_id")
        
        var myArray:[Any] = []
        for item in self.list! {
            myArray.append(item.getProductJson())
        }
        
        dictionary.setValue(myArray, forKey: "list")
        
        return dictionary
    }

}

File: mobile_apps/Android/user/dropo/src/main/java/com/dropo/models/datamodels/Specifications.java
package com.dropo.models.datamodels;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;
import java.util.Objects;

public class Specifications implements Parcelable, Comparable<Specifications> {

    public static final Creator<Specifications> CREATOR = new Creator<Specifications>() {
        @Override
        public Specifications createFromParcel(Parcel source) {
            return new Specifications(source);
        }

        @Override
        public Specifications[] newArray(int size) {
            return new Specifications[size];
        }
    };
    @SerializedName("sequence_number")
    @Expose
    private long sequenceNumber;

    private String chooseMessage;
    private int selectedCount;
    @SerializedName("range")
    @Expose
    private int range;
    @SerializedName("max_range")
    @Expose
    private int maxRange;
    @SerializedName("is_required")
    @Expose
    private boolean isRequired;
    @SerializedName("unique_id")
    @Expose
    private int uniqueId;
    @SerializedName("price")
    @Expose
    private double price;
    @SerializedName("name")
    @Expose
    private Object name;
    @SerializedName("list")
    @Expose
    private List<SpecificationSubItem> list;
    @SerializedName("type")
    @Expose
    private int type;
    @SerializedName("user_can_add_specification_quantity")
    @Expose
    private boolean userCanAddSpecificationQuantity;
    @SerializedName("isParentAssociate")
    @Expose
    private boolean isParentAssociate;
    @SerializedName("isAssociated")
    @Expose
    private boolean isAssociated;
    @SerializedName("modifierId")
    @Expose
    private String modifierId;
    @SerializedName("modifierGroupId")
    @Expose
    private String modifierGroupId;
    @SerializedName("modifierName")
    @Expose
    private String modifierName;
    @SerializedName("modifierGroupName")
    @Expose
    private String modifierGroupName;
    @SerializedName("_id")
    @Expose
    private String id;

    public Specifications() {
    }

    protected Specifications(Parcel in) {
        this.id = in.readString();
        this.range = in.readInt();
        this.maxRange = in.readInt();
        this.isRequired = in.readByte() != 0;
        this.uniqueId = in.readInt();
        this.price = in.readDouble();
        this.name = in.readString();
        this.list = in.createTypedArrayList(SpecificationSubItem.CREATOR);
        this.type = in.readInt();
        this.sequenceNumber = in.readLong();
        this.userCanAddSpecificationQuantity = in.readByte() != 0;
        this.modifierId = in.readString();
        this.modifierGroupId = in.readString();
        this.isAssociated = in.readByte() != 0;
        this.isParentAssociate = in.readByte() != 0;
        this.modifierName = in.readString();
        this.modifierGroupName = in.readString();
    }

    public String getChooseMessage() {
        return chooseMessage;
    }

    public void setChooseMessage(String chooseMessage) {
        this.chooseMessage = chooseMessage;
    }

    public int getSelectedCount() {
        return selectedCount;
    }

    public void setSelectedCount(int selectedCount) {
        this.selectedCount = selectedCount;
    }

    public int getRange() {
        return range;
    }

    public void setRange(int range) {
        this.range = range;
    }

    public int getMaxRange() {
        return maxRange;
    }

    public void setMaxRange(int maxRange) {
        this.maxRange = maxRange;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public String getName() {
        if (name instanceof String) {
            return String.valueOf(name);
        } else {
            return "";
        }
    }

    public void setName(Object name) {
        this.name = name;
    }

    public List<SpecificationSubItem> getList() {
        return list;
    }

    public void setList(List<SpecificationSubItem> list) {
        this.list = list;
    }

    public boolean isUserCanAddSpecificationQuantity() {
        return userCanAddSpecificationQuantity;
    }

    public void setUserCanAddSpecificationQuantity(boolean userCanAddSpecificationQuantity) {
        this.userCanAddSpecificationQuantity = userCanAddSpecificationQuantity;
    }

    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }

    public int getUniqueId() {
        return uniqueId;
    }

    public void setUniqueId(int uniqueId) {
        this.uniqueId = uniqueId;
    }

    public boolean isRequired() {
        return isRequired;
    }

    public void setRequired(boolean required) {
        isRequired = required;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public boolean isParentAssociate() {
        return isParentAssociate;
    }

    public void setParentAssociate(boolean parentAssociate) {
        isParentAssociate = parentAssociate;
    }

    public boolean isAssociated() {
        return isAssociated;
    }

    public void setAssociated(boolean associated) {
        isAssociated = associated;
    }

    public String getModifierId() {
        return modifierId;
    }

    public void setModifierId(String modifierId) {
        this.modifierId = modifierId;
    }

    public String getModifierGroupId() {
        return modifierGroupId;
    }

    public void setModifierGroupId(String modifierGroupId) {
        this.modifierGroupId = modifierGroupId;
    }

    public String getModifierName() {
        return modifierName;
    }

    public void setModifierName(String modifierName) {
        this.modifierName = modifierName;
    }

    public String getModifierGroupName() {
        return modifierGroupName;
    }

    public void setModifierGroupName(String modifierGroupName) {
        this.modifierGroupName = modifierGroupName;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.id);
        dest.writeInt(this.range);
        dest.writeInt(this.maxRange);
        dest.writeByte(this.isRequired ? (byte) 1 : (byte) 0);
        dest.writeInt(this.uniqueId);
        dest.writeDouble(this.price);
        dest.writeString(name instanceof String ? String.valueOf(name) : "");
        dest.writeTypedList(this.list);
        dest.writeInt(this.type);
        dest.writeLong(this.sequenceNumber);
        dest.writeByte(this.userCanAddSpecificationQuantity ? (byte) 1 : (byte) 0);
        dest.writeString(this.modifierId);
        dest.writeString(this.modifierGroupId);
        dest.writeByte(this.isAssociated ? (byte) 1 : (byte) 0);
        dest.writeByte(this.isParentAssociate ? (byte) 1 : (byte) 0);
        dest.writeString(this.modifierName);
        dest.writeString(this.modifierGroupName);
    }

    public long getSequenceNumber() {
        return sequenceNumber;
    }

    @Override
    public int compareTo(Specifications o) {
        return this.sequenceNumber > o.sequenceNumber ? 1 : -1;
    }

    @Override
    public int hashCode() {
        return Objects.hash(sequenceNumber, chooseMessage, selectedCount, range, maxRange, isRequired, uniqueId, price, name, list, type, userCanAddSpecificationQuantity, isParentAssociate, isAssociated, modifierId, modifierGroupId, modifierName, modifierGroupName, id);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Specifications that = (Specifications) o;
        return uniqueId == that.uniqueId && Objects.equals(list, that.list);
    }
}